? browser/app/firefox
? browser/app/firefox-bin
? browser/app/profile/extensions/{972ce4c6-7e08-4474-a285-3208198ce6fd}/install.rdf
? browser/components/build/dependentLibs.h
Index: netwerk/protocol/http/public/nsIHttpProtocolHandler.idl
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/public/nsIHttpProtocolHandler.idl,v
retrieving revision 1.7
diff -d -u -p -6 -r1.7 nsIHttpProtocolHandler.idl
--- netwerk/protocol/http/public/nsIHttpProtocolHandler.idl	14 Nov 2003 20:51:55 -0000	1.7
+++ netwerk/protocol/http/public/nsIHttpProtocolHandler.idl	15 Dec 2004 02:17:14 -0000
@@ -99,12 +99,22 @@ interface nsIHttpProtocolHandler : nsIPr
     attribute ACString language;
 
     /**
      * Get/Set the application comment misc portion.
      */
     attribute ACString misc;
+
+    /**
+     * The application string (e.g. "Firefox")
+     */
+    attribute ACString app;
+
+    /**
+     * The application sub string (e.g. "1.0")
+     */
+    attribute ACString appSub;
 };
 
 %{C++
 /**
  * At initialization time, the HTTP handler will initialize each service
  * registered under this category:
Index: netwerk/protocol/http/src/nsHttpHandler.cpp
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/src/nsHttpHandler.cpp,v
retrieving revision 1.94.2.1.2.3
diff -d -u -p -6 -r1.94.2.1.2.3 nsHttpHandler.cpp
--- netwerk/protocol/http/src/nsHttpHandler.cpp	15 Sep 2004 19:43:43 -0000	1.94.2.1.2.3
+++ netwerk/protocol/http/src/nsHttpHandler.cpp	15 Dec 2004 02:17:15 -0000
@@ -214,12 +214,14 @@ nsHttpHandler::Init()
     LOG(("> app-version = %s\n", mAppVersion.get()));
     LOG(("> platform = %s\n", mPlatform.get()));
     LOG(("> oscpu = %s\n", mOscpu.get()));
     LOG(("> security = %s\n", mSecurity.get()));
     LOG(("> language = %s\n", mLanguage.get()));
     LOG(("> misc = %s\n", mMisc.get()));
+    LOG(("> app = %s\n", mApp.get()));
+    LOG(("> app-sub = %s\n", mAppSub.get()));
     LOG(("> vendor = %s\n", mVendor.get()));
     LOG(("> vendor-sub = %s\n", mVendorSub.get()));
     LOG(("> vendor-comment = %s\n", mVendorComment.get()));
     LOG(("> product = %s\n", mProduct.get()));
     LOG(("> product-sub = %s\n", mProductSub.get()));
     LOG(("> product-comment = %s\n", mProductComment.get()));
@@ -520,12 +522,14 @@ nsHttpHandler::BuildUserAgent()
                            mOscpu.Length() +
                            mLanguage.Length() +
                            mMisc.Length() +
                            mProduct.Length() +
                            mProductSub.Length() +
                            mProductComment.Length() +
+                           mApp.Length() +
+                           mAppSub.Length() +
                            mVendor.Length() +
                            mVendorSub.Length() +
                            mVendorComment.Length() +
                            22);
 
     // Application portion
@@ -563,12 +567,22 @@ nsHttpHandler::BuildUserAgent()
             mUserAgent += " (";
             mUserAgent += mProductComment;
             mUserAgent += ')';
         }
     }
 
+    // App portion
+    if (!mApp.IsEmpty()) {
+        mUserAgent += ' ';
+        mUserAgent += mApp;
+        if (!mAppSub.IsEmpty()) {
+            mUserAgent += '/';
+            mUserAgent += mAppSub;
+        }
+    }
+
     // Vendor portion
     if (!mVendor.IsEmpty()) {
         mUserAgent += ' ';
         mUserAgent += mVendor;
         if (!mVendorSub.IsEmpty()) {
             mUserAgent += '/';
@@ -704,12 +718,23 @@ nsHttpHandler::PrefsChanged(nsIPrefBranc
         prefs->GetCharPref(UA_PREF("appVersion"),
             getter_Copies(mAppVersion));
         if (mAppVersion.IsEmpty())
             mAppVersion.Adopt(nsCRT::strdup(UA_APPVERSION));
         mUserAgentIsDirty = PR_TRUE;
     }
+    if (PREF_CHANGED(UA_PREF("app"))) {
+        prefs->GetCharPref(UA_PREF("app"),
+            getter_Copies(mApp));
+        mUserAgentIsDirty = PR_TRUE;
+    }
+    if (PREF_CHANGED(UA_PREF("appSub"))) {
+        prefs->GetCharPref(UA_PREF("appSub"),
+            getter_Copies(mAppSub));
+        mUserAgentIsDirty = PR_TRUE;
+    }
+
 
     // Gather vendor values.
     if (PREF_CHANGED(UA_PREF("vendor"))) {
         prefs->GetCharPref(UA_PREF("vendor"),
             getter_Copies(mVendor));
         mUserAgentIsDirty = PR_TRUE;
@@ -1448,18 +1473,50 @@ NS_IMETHODIMP
 nsHttpHandler::GetAppVersion(nsACString &value)
 {
     value = mAppVersion;
     return NS_OK;
 }
 
+
+NS_IMETHODIMP
+nsHttpHandler::GetApp(nsACString &value)
+{
+    value = mApp;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::GetAppSub(nsACString &value)
+{
+    value = mAppSub;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::SetApp(const nsACString &value)
+{
+    mApp = value;
+    mUserAgentIsDirty = PR_TRUE;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::SetAppSub(const nsACString &value)
+{
+    mAppSub = value;
+    mUserAgentIsDirty = PR_TRUE;
+    return NS_OK;
+}
+
 NS_IMETHODIMP
 nsHttpHandler::GetVendor(nsACString &value)
 {
     value = mVendor;
     return NS_OK;
 }
+
 NS_IMETHODIMP
 nsHttpHandler::SetVendor(const nsACString &value)
 {
     mVendor = value;
     mUserAgentIsDirty = PR_TRUE;
     return NS_OK;
@@ -1468,12 +1525,13 @@ nsHttpHandler::SetVendor(const nsACStrin
 NS_IMETHODIMP
 nsHttpHandler::GetVendorSub(nsACString &value)
 {
     value = mVendorSub;
     return NS_OK;
 }
+
 NS_IMETHODIMP
 nsHttpHandler::SetVendorSub(const nsACString &value)
 {
     mVendorSub = value;
     mUserAgentIsDirty = PR_TRUE;
     return NS_OK;
Index: netwerk/protocol/http/src/nsHttpHandler.h
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/src/nsHttpHandler.h,v
retrieving revision 1.37.2.1
diff -d -u -p -6 -r1.37.2.1 nsHttpHandler.h
--- netwerk/protocol/http/src/nsHttpHandler.h	8 May 2004 00:37:08 -0000	1.37.2.1
+++ netwerk/protocol/http/src/nsHttpHandler.h	15 Dec 2004 02:17:15 -0000
@@ -264,12 +264,14 @@ private:
     nsXPIDLCString mVendor;
     nsXPIDLCString mVendorSub;
     nsXPIDLCString mVendorComment;
     nsXPIDLCString mProduct;
     nsXPIDLCString mProductSub;
     nsXPIDLCString mProductComment;
+    nsXPIDLCString mApp;
+    nsXPIDLCString mAppSub;
 
     nsCString      mUserAgent;
     nsXPIDLCString mUserAgentOverride;
     PRPackedBool   mUserAgentIsDirty; // true if mUserAgent should be rebuilt
 
     PRPackedBool   mUseCache;
Index: browser/app/profile/firefox.js
===================================================================
RCS file: /cvsroot/mozilla/browser/app/profile/firefox.js,v
retrieving revision 1.7.4.40
diff -d -u -p -6 -r1.7.4.40 firefox.js
--- browser/app/profile/firefox.js	5 Nov 2004 09:10:05 -0000	1.7.4.40
+++ browser/app/profile/firefox.js	15 Dec 2004 02:17:15 -0000
@@ -119,14 +119,14 @@ pref("update.severity", 0); 
 pref("xpinstall.whitelist.add", "update.mozilla.org");
 
 pref("keyword.enabled", true);
 pref("keyword.URL", "http://www.google.com/search?btnI=I%27m+Feeling+Lucky&ie=UTF-8&oe=UTF-8&q=");
 
 pref("general.skins.selectedSkin", "classic/1.0");
-pref("general.useragent.vendor", "Firefox");
-pref("general.useragent.vendorSub", 
+pref("general.useragent.app", "Firefox");
+pref("general.useragent.appSub", 
 #expand __APP_VERSION__
 );
 
 pref("general.smoothScroll", false);
 #ifdef XP_UNIX
 pref("general.autoScroll", false);
